### DISCUSSION TITLE
Homeassistant Energy Dashboard not working

### DISCUSSION THREAD

**Discussion by komax74** [jomjol/AI-on-the-edge-device#1847] [General]
### The Problem

After updating to 13.0.8 (from 12.0.1) with the home assist discovery function, I can no longer update the Energy Dashboard (I use it for measure gas)
I've read and reread your posts and the wiki but I couldn't figure out what I need to do, if I go to settings > dashboard > energy I only see the old sensor (which it says is no longer connected) but if I do add I have no entities (neither gas nor water).
Who can help me step by step? The setting is correct, on mqtt I see the system and it imports the data into the main dashboard so the problem is only the energy dashboard settings, one which by the way to not lose the history should update the old data and not make new data.

Thanks and sorry if a duplicate post but I can't find solutions.

### Version

13.0.8

### Logfile

```shell
[1d01h10m50s] 2023-01-15T11:52:38	<WRN>	[OTA] Reboot in 5sec
[0d00h00m04s] 2023-01-15T09:45:46	<INF>	[MAIN] =================================================
[0d00h00m04s] 2023-01-15T09:45:46	<INF>	[MAIN] ==================== Startup ====================
[0d00h00m04s] 2023-01-15T09:45:46	<INF>	[MAIN] =================================================
[0d00h00m04s] 2023-01-15T09:45:46	<INF>	[MAIN] Branch: 'HEAD', Revision: 04f2f23+, Date/Time: 2022-12-19 20:41, Web UI: Release: v13.0.8 (Commit: 04f2f23+) 
[0d00h00m04s] 2023-01-15T09:45:46	<INF>	[MAIN] Reset reason: Via esp_restart
[0d00h00m07s] 2023-01-15T09:45:49	<INF>	[MAIN] =================================================
[0d00h00m07s] 2023-01-15T09:45:49	<INF>	[MAIN] ================== Main Started =================
[0d00h00m07s] 2023-01-15T09:45:49	<INF>	[MAIN] =================================================
[0d00h00m07s] 2023-01-15T09:45:49	<INF>	[MAIN] The device has 8 MBytes of PSRAM
[0d00h00m09s] 2023-01-15T09:45:51	<INF>	[MAIN] Initialization completed successfully!
```

### Expected Behavior

Find the entity to replace the one generated with the old firmware to continue to see the graphs in energy dashboard

<img width="343" alt="Screenshot 2023-01-15 alle 12 08 40" src="https://user-images.githubusercontent.com/4668800/212537264-dbb9e35d-65e9-479f-abff-33c126e90b4e.png">

**Comment by caco3** [comment]
According to the screenshots, Homeassistant MQTT Discovery works for you and you see the device under `http://<HOMEASSISTANT IP>/config/integrations -> MQTT`.

If you now click on the top (Sensori) frame onto `Value` and there onto the 2nd tab (Settings), you should see the Entity ID.
![grafik](https://user-images.githubusercontent.com/1783586/212546688-0baa16e9-e277-4a33-994b-4765e84a8646.png)
Copy that entry and add it to your Energy Dashboard (`http://<HOMEASSISTANT IP>/config/energy`) as a water meter:
![grafik](https://user-images.githubusercontent.com/1783586/212546737-3d7b4d69-25b1-4c77-8f38-b46271cae8a2.png)

If this is not possible, I think it is an issue on your Homeassistant side and you should ask for support in their forum.

**Comment by komax74** [comment]
Not work, in the MQTT there is correct information for the sensor, the name is sensor.gasmeter_value_2 but when go to energy isn't visible.
I try water and gas, there are only the old sensor but not working anymore.
[](url)
<img width="614" alt="Screenshot 2023-01-16 alle 22 25 33" src="https://user-images.githubusercontent.com/4668800/212768234-63dd1d8e-0abc-4389-9777-bacbfd0592c1.png">
<img width="519" alt="Screenshot 2023-01-16 alle 22 27 29" src="https://user-images.githubusercontent.com/4668800/212768251-7d0b6851-4e4d-4a1e-9aed-fcfc6d837f23.png">
<img width="480" alt="Screenshot 2023-01-16 alle 22 26 45" src="https://user-images.githubusercontent.com/4668800/212768254-785b51c6-5320-4a8c-8be4-eb5c0138aaa9.png">

**Reply by caco3** [reply]
I am confident that it is an Issue on Homeassistant side.
You could test it with a new empty Homeassistant installation for comparison.
I also had issues before with other sensors which did not show up anymore after I played around.

I suggest to go to the Homeassistant Forum to get support.

**Comment by rolandsteinmeyer** [comment]
Check If your sensor hast correct device_class and state_class.
